src/runtime/HalideBuffer.h

```diff
@@ -1070,6 +1094,7 @@ class Buffer {
     void add_dimension() {
         // Check there's enough space for a new dimension.
         assert(dims < D);
+        disown_device_handle();
```

Comment:
Isn't this actually OK? I think it's only once you do something to change the new dimension that the buffer needs to be disowned.